### ISSUE TITLE
feat: post-training data support — fine-tune on ClawBench V1/V2 traces

### ISSUE THREAD

**Issue by reacher-z** [TIGER-AI-Lab/ClawBench#158]
## Motivation

ClawBench ships **2,326 graded browser-agent trajectories** across V1 (153 tasks × 6 frontier models = 918) and V2 (130 tasks × 7 model×harness combinations = 806 + ongoing). Each run is a 5-layer bundle (recording.mp4 / actions.jsonl / agent-messages.jsonl / requests.jsonl / interception.json / run-meta.json) with a deterministic Stage-1 grade and an LLM-judge Stage-2 grade.

That's already useful **as an eval set** — the leaderboard. But the same data is one `jq` away from being a **fine-tuning corpus**, and we should make that path frictionless. Use cases the community has been asking for:

- **SFT** on Claude Sonnet 4.6 / Opus 4.6 / Opus 4.7 trajectories (the closed-frontier teachers) — train your own open-weight student that drives a browser
- **Process Reward Models (PRMs)** from per-step action grades (which step caused the failure?)
- **DPO / preference learning** from success-vs-failure pairs on identical tasks across 13 models
- **Distillation** with the V2Trace bundles as multi-modal teacher demonstrations (`recording.mp4` for VLM + `actions.jsonl` for tool-use grounding)
- **Judge fine-tuning** — train an open-weight judge from our `deepseek-v4-pro` Stage-2 labels

## What to try

This is a **research spike** issue — explore feasibility, prototype one workflow end-to-end, decide later whether to land it as a permanent feature. Three angles to spike, pick one and report findings:

### Spike 1 — \`clawbench-eval-sft\` (SFT-ready JSONL converter)

Add a CLI subcommand:

```bash
clawbench dataset --corpus v1 --models claude-sonnet-4-6 \\
  --format sharegpt --filter pass=true --out v1-sonnet-pass-only.jsonl
```

Outputs a flat JSONL where each row is a complete trajectory in one of:
- **ShareGPT** format (`{conversations: [{from: system|human|gpt|tool, value: ...}], ...}`) — broadly compatible with HF SFT trainers
- **Anthropic messages** format (`{messages: [{role, content}], ...}`) — easy round-trip to Claude
- **Custom** ClawBench format with `actions[].step_id` cross-references back to the recording

Surfaces hidden curation knobs: filter by reward, by metaclass, by harness, by attempted-task-count.

### Spike 2 — `clawbench-eval-prm` (process reward signal exporter)

ClawBench's per-task `interception.json` is a final-outcome grade; the trajectory has many intermediate steps. To support PRM training:

- Walk `actions.jsonl` + `agent-messages.jsonl` chronologically
- Heuristic per-step score (or LLM-judged: same `deepseek-v4-pro` judge but re-prompted per turn)
- Output a `{step_id, action, reasoning, step_reward, terminal_reward}` JSONL ready for AgentPRM-style training

### Spike 3 — `clawbench-eval-pairs` (DPO success/failure pair miner)

For each task that has both a passing and failing trajectory from different models:
- Pair them up `(chosen, rejected)`
- Output a flat JSONL ready for `trl.DPOTrainer` / `LLaMA-Factory` DPO pipelines

130 V2 tasks × ~7 models = ~3000 attempted (run, task) pairs; rough back-of-envelope ~500-1,000 contrastive pairs after filtering for has-both-pass-and-fail.

## Implementation hints

- Reuse the existing `scripts/clawbench_rescore.py` parquet+jsonl readers
- Output goes to a new HF dataset: `TIGER-Lab/ClawBench-SFT` / `ClawBench-PRM` / `ClawBench-DPO` (one per spike, only if the spike pans out)
- Keep the **raw trace bundles** untouched — derived datasets are downstream

## Out of scope

- Actually training a model on the derived datasets (separate `harness-bench` style follow-up)
- VLM-specific (image-conditioned) training data — needs frame-extraction pipeline, not in this spike

## Acceptance criteria (per spike)

- [ ] Working CLI subcommand under `clawbench-eval` that runs on a sample of V1 or V2 traces
- [ ] One small derived dataset uploaded to HF (e.g., 200 trajectories)
- [ ] Brief writeup in `docs/training/` covering format, filtering knobs, gotchas, and what's missing for full-scale training

**Comment by Perry2004** [comment]
There's already guide on this: https://github.com/TIGER-AI-Lab/ClawBench/blob/main/docs/prorl-rl-env.md. 
Actual training codes should not go into this repo.
